tests/kernel/interrupt/src/dynamic_isr.c

```diff
@@ -133,12 +114,13 @@ extern const void *x86_irq_args[];
 			"parameter(0x%lx) in handler is not correct",
 			handler_test_result);
 
-	irq_disable(TRIGGER_IRQ_DYN_LINE);
+	trigger_irq(vector_num);
 
-	/**TESTPOINT: interrupt cannot be triggered when disable it */
-	zassert_equal(handler_has_run, 1,
-			"interrupt handler should not be triggered again(%d)",
+	/**TESTPOINT: interrupt triggered again */
+	zassert_equal(handler_has_run, 2,
+			"interrupt triggered but handler has not run(%d)",
 			handler_has_run);
+
```

Comment:
This changes the nature of this test. The original makes sure IRQ cannot be triggered after it is disabled. But the new code try to see if it can be triggered again.

Reply:
Hi @dcpleung  sorry for missing this message, This indeed change the logic for testing, but the original test is not totally correct in this cases. Now I it seems like I cannot reopen this PR. So I make some change and start a new PR for it, please refer https://github.com/zephyrproject-rtos/zephyr/pull/40843, could you please help to review it again when you got a chance, thank you!